src/views/Vote/CreateProposal/CreateProposal.tsx

```diff
@@ -0,0 +1,172 @@
+/** @jsxImportSource theme-ui */
+import React, { useState } from 'react'
+import { Box, Flex, Text } from 'theme-ui'
+import { Svg, ArrowBackIcon, Link, Textarea, Button, Spinner, TooltipBubble, Input } from '@ape.swap/uikit'
+import { create } from 'ipfs-http-client'
+import { MarkdownIcon } from 'components/Icons'
+import { styles } from './styles'
+import Preview from './Preview'
+
+export interface InputProps {
+  onChange: (e: React.FormEvent<HTMLInputElement>) => void
+  placeholder?: string
+  value: string
+}
+
+const CreateProposal = () => {
+  const [proposalData, setPropposalData] = useState({
+    title: '',
+    dicussion: '',
+    textarea: '',
+  })
+
+  const [state, setState] = useState({
+    preview: true,
+    loader: false,
+  })
+
+  const InputCreateProposal: React.FC<InputProps> = ({ onChange, placeholder, value }) => {
+    return (
+      <Box sx={styles.inputMain}>
+        <Input sx={styles.input} placeholder={placeholder} value={value} onChange={onChange} width="100%" />
+      </Box>
+    )
+  }
+
+  const handleChange = (e) => {
+    setPropposalData({ ...proposalData, title: e.target.value })
+  }
+  const handleChangeDiscussion = (e) => {
+    setPropposalData({ ...proposalData, dicussion: e.target.value })
+  }
+  const handleTextArea = (e) => {
+    if (e.target.value.length <= 14400) {
+      setPropposalData({
+        ...proposalData,
+        textarea: e.target.value,
+      })
+    }
+  }
+
+  const handleDrop = async (e) => {
+    e.preventDefault()
+    e.stopPropagation()
+
+    setState({ ...state, loader: true })
+    const imageFile = e.dataTransfer.files[0]
+    const client = create({ url: 'https://ipfs.infura.io:5001/api/v0' })
+    try {
+      const added = await client.add(imageFile)
+
+      if (e.target.value.length <= 14400) {
+        const url = `https://ipfs.infura.io/ipfs/${added.path}`
+        setPropposalData((prevValue) => ({
+          ...prevValue,
+          textarea: `${prevValue.textarea} \n![${imageFile?.name}](${url})\n`,
+        }))
+      }
+    } catch (error) {
+      console.log('Error uploading file: ', error)
+    } finally {
+      setState({ ...state, loader: false })
+    }
+  }
+
+  const handleDragOver = (e) => {
+    e.preventDefault()
+  }
+
+  const LoaderAndtext = () => {
```

Comment:
This can be put in the code directly, no need to abstract it.

Reply:
Done